### PR TITLE
feat: implement FormatPatterns

### PR DESCRIPTION
**Pull request by Fyko**
This PR creates a common object of message formats as regexes.
https://discord.com/developers/docs/reference#message-formatting-formats

### CODE REVIEW

File: common/index.ts [hunk 1]
```diff
@@ -164,3 +164,13 @@ export type Snowflake = `${bigint}`;
  * @internal
  */
 export type Permissions = `${bigint}`;
+
+/**
+ * https://discord.com/developers/docs/reference#message-formatting-formats
+ */
+export const FormattingPatterns = {
+	User: /<@!?(\d{17,19})>/,
```
Comment (vladfrangu) [inline common/index.ts:172]: Should contain the 3 possibilities we'd want - User, user w nick and user with optional nick

File: common/index.ts [hunk 2]
```diff
@@ -164,3 +164,13 @@ export type Snowflake = `${bigint}`;
...
+	Channel: /<#(\d{17,19})>/,
+	Role: /<@&(\d{17,19})>/,
+	Emoji: /<?(a)?:?(\w{2,32}):(\d{17,19})>?/,
```
Comment (vladfrangu) [inline common/index.ts:175]: Please split this into 3 (non animated, only animated, both)

File: common/index.ts [hunk 3]
```diff
@@ -164,3 +164,13 @@ export type Snowflake = `${bigint}`;
...
+};
```
Comment (vladfrangu) [inline common/index.ts:176]: cast as const

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: common/index.ts [hunk 4]
```diff
@@ -169,8 +169,14 @@ export type Permissions = `${bigint}`;
...
+	UserWithNickname: /<@!(\d{17,19})>/,
+	UserWithOptionalNickname: /<@!?(\d{17,19})>/,
+
 	Channel: /<#(\d{17,19})>/,
 	Role: /<@&(\d{17,19})>/,
+
 	Emoji: /<?(a)?:?(\w{2,32}):(\d{17,19})>?/,
-};
+	AnimatedEmoji: /<a:(\w{2,32}):(\d{17,19})>?/,
+	StaticEmoji: /<:(\w{2,32}):(\d{17,19})>?/,
```
Comment (vladfrangu) [inline common/index.ts:181]: Include the d.js one with the match groups

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: common/index.ts [hunk 5]
```diff
@@ -164,3 +164,19 @@ export type Snowflake = `${bigint}`;
...
+	User: /<@(\d{17,19})>/,
+	UserWithNickname: /<@!(\d{17,19})>/,
+	UserWithOptionalNickname: /<@!?(\d{17,19})>/,
+
+	Channel: /<#(\d{17,19})>/,
+	Role: /<@&(\d{17,19})>/,
+
+	Emoji: /<?(a)?:?(\w{2,32}):(\d{17,19})>?/,
+	AnimatedEmoji: /<(a):(\w{2,32}):(\d{17,19})>?/,
+	StaticEmoji: /<:(\w{2,32}):(\d{17,19})>?/,
```
Comment (vaporox) [inline common/index.ts:181]:
```suggestion
	Emoji: /<(a)?:(\w{2,32}):(\d{17,19})>/,
	AnimatedEmoji: /<(a):(\w{2,32}):(\d{17,19})>/,
	StaticEmoji: /<:(\w{2,32}):(\d{17,19})>/,
```
those aren't optional for the other mention patterns either

File: common/index.ts [hunk 6]
```diff
@@ -164,3 +164,19 @@ export type Snowflake = `${bigint}`;
  * @internal
  */
 export type Permissions = `${bigint}`;
+
+/**
+ * https://discord.com/developers/docs/reference#message-formatting-formats
+ */
+export const FormattingPatterns = {
+	User: /<@(\d{17,19})>/,
```
Comment (vladfrangu) [inline common/index.ts:172]: add docs for what each represent :3

**Review by vladfrangu — CHANGES_REQUESTED** [review]

**Review by vladfrangu — APPROVED** [review]

**Pull request merged**